api/capi/src/nnstreamer-capi-single.c

```diff
@@ -160,6 +160,9 @@ __process_output (ml_single * single_h, GstTensorMemory * out_tensors)
     if (status != ML_ERROR_NONE) {
       ml_loge ("Failed to allocate the memory block.");
       (*single_h->output) = NULL;
+      /** free computed output data */
+      for (i = 0; i < single_h->out_info.num_tensors; i++)
```

Comment:
To reduce duplicated code (line 175~180), how about adding goto and checking the flag before return?
```
gboolean need_free = false;
...
process_error:
if (need_free)
  //free mem
```

Reply:
Done.